Fix/8859 prishe benediction promathia

Alphiez opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Closes #8859.

Fixes Dawn (CoP 8-4) behavior where Prishe’s Benediction could incorrectly heal Promathia. While validating the fix, I found two related Dawn issues (Raise targeting on NPC allies and Prishe reusing Daedalus Wing) and addressed them as well.

### Technical notes / rationale

#### 1) Prishe Benediction incorrectly healing Promathia (Issue #8859)
- Root cause: Benediction’s healing logic path could select Promathia under certain conditions (enemy being treated as a valid recipient / missing ally-only restriction).
- Fix: Restrict Benediction to apply only to valid friendly targets (party/allies) in this battlefield context, preventing Promathia from being eligible for the heal.

#### 2) Raise spells failing on Prishe (NPC ally) [extra fix]
- Root cause: Raise spell scripts used `target:hasRaiseTractorMenu()` as a universal block. In the image below, if you look at "RAISE CHECK", we can see that the menu is set to TRUE - That check is only meaningful for PCs (playable characters) (Raise/Tractor menu state), but it caused NPC allies to fail the casting check even when dead.
```
printf("RAISE CHECK: name=%s hp=%d dead=%s alive=%s menu=%s pool=%d obj=%d",
        name, hp, tostring(dead), tostring(alive), tostring(menu), pool, obj)
```
<img width="1239" height="243" alt="image" src="https://github.com/user-attachments/assets/67416f80-54e8-4462-8773-204686c6fc85" />
- Fix: Only apply `hasRaiseTractorMenu()` gating to PCs. For non-PC targets, require that the target is explicitly raisable via the behavior bit (`BEHAVIOR_RAISABLE`) before allowing the cast.
- Result: Players remain unchanged (normal Raise menu rules), while raisable NPC allies can be targeted by Raise I/II/III.

#### 3) Prishe repeatedly using Daedalus Wing [extra fix]
- Root cause: `useMobAbility(1487)` was triggered on engage every time she re-engaged (including after being raised / phase transitions).
- Fix: Track a local var (`used_item_1`) and only allow the ability once. Initialize it at battlefield spawn and set it so Phase II / resets don’t allow re-use.
- Result: Prishe uses Daedalus Wing once at the start of combat, and never again during the battlefield session.

## Steps to test these changes

1. Grant the Dawn mission: !addmission 6 840
2. Zone into Empyreal Paradox: !zone 36
3. Walk forward and enter the Dawn battlefield.
4. Verify Benediction does not heal Promathia: by using !hp <low_hp#> to force benediction or just wait
5. Verify Raise I/II/III on Prishe: with !hp 0 the try to raise her with a Raise spell
- If there is an issue here, it could be due to the db of prishe not having the correct behavior. I don't remember updating the data base to reflect the correct behavior - I think it exists as it should but here are steps to check
- Execute mariadb.exe (for me its located in C:\Program Files\MariaDB 10.6\bin>)  .\mariadb.exe -u root -p
- USE xidb;
- Use the following Query: ```SELECT poolid,name,behavior,(behavior & 4) AS raisable FROM mob_pools WHERE poolid=3199;```
- You should see Prishe behavior is 5 thus her raisable value is 4 making her a raisable target.
```
MariaDB [xidb]> SELECT poolid,name,behavior,(behavior & 4) AS raisable FROM mob_pools WHERE poolid=3199;
+--------+--------+----------+----------+
| poolid | name   | behavior | raisable |
+--------+--------+----------+----------+
|   3199 | Prishe |        5 |        4 |
+--------+--------+----------+----------+
1 row in set (0.000 sec)
```
6. Verify Daedalus Wing is one-time only:
- On initial engage, confirm Prishe uses Daedalus Wing once.
- Let her die and raise her again.
- Confirm she does NOT use Daedalus Wing again (including in Phase II engage).

Alphiez committed:
Fixed Prishe Benediction healing Promathia *REF 8859*

Alphiez committed:
fix(spells): allow Raise spells to target raisable NPC allies

Alphiez committed:
fix(cop): prevent Prishe from reusing Daedalus Wing in Dawn

Xaver-DaRed commented on scripts/actions/mobskills/benediction.lua:
```diff
@@ -9,6 +9,11 @@ mobskillObject.onMobSkillCheck = function(target, mob, skill)
+
+    if target:getAllegiance() ~= mob:getAllegiance() then
+        return 0
+    end
```
This kind of behavior/logic belongs to core, target-finding, not here.

Xaver-DaRed commented on scripts/actions/mobskills/benediction.lua:
```diff
@@ -21,4 +26,4 @@ mobskillObject.onMobWeaponSkill = function(target, mob, skill)
 end
 
-return mobskillObject
+return mobskillObject
```
Newline t the end of file is needed.

Xaver-DaRed commented on scripts/actions/spells/white/raise.lua:
```diff
@@ -3,33 +3,41 @@
 -----------------------------------
 ---@type TSpell
 local spellObject = {}
+local BEHAVIOR_RAISABLE = 0x004
```
This is unneeded. We already have a global, predefined enum for this
`xi.behavior.RAISABLE`

Xaver-DaRed commented on scripts/actions/spells/white/raise_ii.lua:
```diff
@@ -1,35 +1,43 @@
 -----------------------------------
 ---@type TSpell
 local spellObject = {}
+local BEHAVIOR_RAISABLE = 0x004
```
This is unneeded. We already have a global, predefined enum for this
`xi.behavior.RAISABLE`

Xaver-DaRed commented on scripts/actions/spells/white/raise_iii.lua:
```diff
@@ -1,35 +1,43 @@
 -----------------------------------
 ---@type TSpell
 local spellObject = {}
+local BEHAVIOR_RAISABLE = 0x004
```
This is unneeded. We already have a global, predefined enum for this
`xi.behavior.RAISABLE`

Xaver-DaRed commented on scripts/battlefields/Empyreal_Paradox/dawn.lua:
```diff
@@ -66,8 +68,13 @@ function content:onEventFinishBattlefield(player, csid, option, npc)
+        end
         ally:setPos(spawn.x, spawn.y, spawn.z, spawn.rot)
     end
+    
```
None of this is needed, I'm pretty sure. You are already gating its usage to once per battlefiled by checking the variable

Alphiez replied:
Thank you, I made the changes and improved Prishe.lua logic to handle the one time use.

Xaver-DaRed commented on scripts/zones/Empyreal_Paradox/mobs/Prishe.lua:
```diff
@@ -45,7 +45,12 @@ entity.onMobRoam = function(mob)
-    mob:useMobAbility(1487)
+    -- Logic to only allow Daedalus Wing to be cast once
+    if mob:getLocalVar('used_item_1') == 0 then
+        mob:setLocalVar('used_item_1', 1)
```
'used_item_1' -> 'usedWing'

Alphiez replied:
changed to usedWing, thanks.

Xaver-DaRed commented on scripts/actions/spells/white/raise_iii.lua:
```diff
@@ -1,35 +1,43 @@
+        if bit.band(behavior, BEHAVIOR_RAISABLE) == 0 then
+            return xi.msg.basic.MAGIC_CANNOT_BE_CAST
+        end
+    end
```
```lua
-- Non-PC targets must be explicitly raisable (behavior bit)
if not target:isPC() and bit.band(target:getBehavior(), xi.behavior.RAISABLE) == 0 then
    return xi.msg.basic.MAGIC_CANNOT_BE_CAST
end
```

Alphiez committed:
Fix Prishe Lua Logic, remove Logic in Dawn lua for Daedalus wing

Alphiez committed:
Fix self-centered AoE targeting to respect allegiance

Alphiez committed:
Revert Benediction Lua workaround now handled in core

Alphiez committed:
Fix Prishe engage script block structure

Alphiez committed:
Raise: enforce RAISABLE for non-PC targets

Alphiez committed:
Prishe: clean up onMobEngage logic

Alphiez committed:
Benediction: remove debug and follow style rules

Alphiez commented:
Apologies for the extra commits, I was getting familiar with the sanity check workflow and figured how to run it from the terminal so I could verify everything passed before pushing.

**Summary of changes:**
- Fixed core (targetfind.cpp) AoE target selection so self-centered abilities (like Prishe’s Benediction) respect allegiance and no longer include enemy entities such as Promathia.
- Removed the Lua-level Benediction workaround since the behavior is now correctly enforced in core targeting logic.
- Updated Raise / Raise II / Raise III to consistently require the `xi.behavior.RAISABLE` behavior flag for non-PC targets.
- Cleaned up Prishe and Dawn Lua scripts (Daedalus Wing usage and formatting).
- Cleaned up white spaces, missing newlines at end, etc.

All changes now pass sanity checks

Skold177 commented on scripts/zones/Empyreal_Paradox/mobs/Prishe.lua:
```diff
@@ -45,7 +45,17 @@ entity.onMobRoam = function(mob)
+
+    if battlefield:getLocalVar('usedWing') ~= 1 then
+        battlefield:setLocalVar('usedWing', 1)
+        mob:useMobAbility(1487)
```
Whenever you're working on something and notice hard coded IDs like this you can (and should) clean them up with a enum entry for improved readability.

Check out mob_skill.lua and add the entry there. something like 1487 = PRISHE_DAEDALUS_WING, then you can clean up that line to mob:useMobAbility(xi.mobSkill.PRISHE_DAEDALUS_WING)

Xaver-DaRed replied:
The naming convention for mobskills is
`MOBSKILL_NAME_USE_CASE`
In this case it would be
`DAEDALUS_WING_COP_PRISHE`

Alphiez replied:
Thank you for the feedback, I agree. I'll take a look tonight and make this change. Appreciate the mentorship and patience.

Alphiez committed:
Add Wing to the mob_skill.lua and Remove hard code 1487 from prishe.lua

Alphiez committed:
Revert my changesit broke enemy mob aoe
